eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/RequestPathSegment.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Generator.Management.Models
+{
+    internal class RequestPathSegment : IEquatable<RequestPathSegment>
+    {
+        private readonly string _value;
+        private readonly bool _isConstant;
+        private readonly string? _variableName;
+
+        // TODO - scope not supported yet. we need to add support for it in the future.
+        public RequestPathSegment(string value)
+        {
+            _value = value;
+            (_isConstant, _variableName) = ParseValue(value);
+        }
+
+        public bool Equals(RequestPathSegment? other)
+        {
+            if (other is null)
+                return false;
+            return _value == other._value;
+        }
+
+        /// <summary>
+        /// Returns the literal value of this segment.
+        /// For a variable, it would return a string with the curly braces.
+        /// </summary>
+        public string Value => _value;
+
+        /// <summary>
+        /// Returns true if this segment is a constant segment (i.e., it is not enclosed in curly braces).
+        /// </summary>
+        public bool IsConstant => _isConstant;
+
+        /// <summary>
+        /// Returns the variable name if this segment is a variable segment (i.e., it is enclosed in curly braces).
+        /// Throws an exception if this segment is a constant segment.
+        /// </summary>
+        public string VariableName => _variableName ?? throw new InvalidOperationException("This segment is not a variable segment.");
+
+        /// <summary>
+        /// Returns true if this segment is the "providers" segment.
+        /// </summary>
+        public bool IsProvidersSegment => _value.Equals("providers");
+
+        private static (bool IsConstant, string? VariableName) ParseValue(string value)
```

Comment:
consider to convert to `bool TryParseVariableName(string value, [NotNullWhen(true)] out string? variableName)`